server/expression.js

```diff
@@ -199,14 +215,15 @@ expression.listMatrix_ = function(expressionPath) {
 };
 
 /**
- * Reads expression matrix from text file.
- * @param {string} expressionFile Path to the expression file.
+ * Reads matrix from text file, returns the whole matrix when input genes and
+ * conditions are empty.
+ * @param {string} fileName Path to the matrix file.
  * @param {!Array<string>} geneNames Names for gene selection.
```

Comment:
You can remove the non-null indicator "!" and let geneNames=null indicate all selection.